src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java

```diff
@@ -451,6 +481,200 @@ protected final void verify(Checker checker,
+                final String actualViolationFileName = actualViolation[0];
+                final String actualViolationMessage = actualViolation[1];
+
+                actualViolations
+                        .computeIfAbsent(actualViolationFileName, key -> new ArrayList<>())
+                        .add(actualViolationMessage);
+            }
+
+            return actualViolations;
+        }
+    }
+
+    /**
+     * Creates a google configuration based on the list of modules.
+     *
+     * @param listOfModules list of modules to load from google config.
+     * @return google configuration based on the list of modules.
+     */
+    private static DefaultConfiguration createGoogleConfig(Map<String, String[]> listOfModules) {
+        final Configuration masterGoogleConfig = getMasterGoogleConfig();
+
+        final DefaultConfiguration root = new DefaultConfiguration(ROOT_MODULE_NAME);
+        final DefaultConfiguration treeWalkerConfig =
+                new DefaultConfiguration(TreeWalker.class.getSimpleName());
+
+        for (Map.Entry<String, String[]> entry : listOfModules.entrySet()) {
+            if (CHECKER_CHILDREN.contains(entry.getKey())) {
+                handleCheckerChildren(masterGoogleConfig, root, entry);
+            }
+            else {
+                handleTreeWalkerChildren(masterGoogleConfig, treeWalkerConfig, entry);
+            }
+        }
+
+        if (treeWalkerConfig.getChildren().length > 0) {
+            root.addChild(treeWalkerConfig);
+        }
+        return root;
+    }
+
+    /**
+     * Returns the whole google configuration.
+     *
+     * @return the google config present in the google_checks.xml
+     * @throws IllegalStateException if there is a problem loading the configuration.
+     */
+    private static Configuration getMasterGoogleConfig() {
+        final Configuration masterGoogleConfig;
+        try {
+            masterGoogleConfig =
+                    ConfigurationLoader.loadConfiguration("/google_checks.xml",
+                            new PropertiesExpander(System.getProperties()));
+        }
+        catch (CheckstyleException ex) {
+            throw new IllegalStateException(ex);
+        }
+        return masterGoogleConfig;
+    }
+
+    /**
+     * Handles the children of the Checker for google config.
+     *
+     * @param masterGoogleConfig the master google config.
+     * @param root the root configuration.
+     * @param entry the entry to handle.
+     */
+    private static void handleCheckerChildren(Configuration masterGoogleConfig,
+           DefaultConfiguration root, Map.Entry<String, String[]> entry) {
+        if (entry.getValue() != null) {
+            final String[] moduleIds = entry.getValue();
+            for (String moduleId : moduleIds) {
+                final Configuration moduleConfig =
+                        getModuleConfig(masterGoogleConfig, entry.getKey(), moduleId);
+                root.addChild(moduleConfig);
```

Comment:
please move this operation outside of method, no reason to provide root as param, and rename method to getCheckerChildren

Reply:
same for handleTreeWalkerChildren

Reply:
Done. Here's the new changes:

https://github.com/checkstyle/checkstyle/blob/a0f82691156c959fbb7dccd991c3e5c4f83022e1/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L620-L631

https://github.com/checkstyle/checkstyle/blob/a0f82691156c959fbb7dccd991c3e5c4f83022e1/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L641-L652

https://github.com/checkstyle/checkstyle/blob/a0f82691156c959fbb7dccd991c3e5c4f83022e1/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L570-L585

this may be not efficient, I'm not sure if this is the exact changes you requested. Please let me know if I still need to change something.